detection_rules/version_lock.py

```diff
@@ -272,11 +281,18 @@ def log_changes(r, route_taken, new_rule_version, *msg):
                     #       We can still inspect the version lock manually after locks are made,
                     #       since it's a good summary of everything that happens
 
-                    # if version bump collides with future bump, fail
-                    # if space, change and log
+                    previous_entry = lock_from_file["previous"][str(min_stack)]
+                    max_allowable_version = previous_entry['max_allowable_version']
+
+                    # if version bump collides with future bump: fail
+                    # if space: change and log
                     info_from_rule = (lock_from_rule['sha256'], lock_from_rule['version'])
-                    info_from_file = (lock_from_file["previous"][str(min_stack)]['sha256'],
-                                      lock_from_file["previous"][str(min_stack)]['version'])
+                    info_from_file = (previous_entry['sha256'], previous_entry['version'])
+
+                    if lock_from_rule['version'] > max_allowable_version:
```

Comment:
Should this be `lock_from_file['version']`? We want to check if the previous entry fork is greater than max allowable version and if so, there is no more buffer space?

Reply:
This makes sure that the rule version has not incremented past the max allowable, which is from the lock version